backend/app/core/security.py

```
@@ -241,4 +247,105 @@ def get_current_user_from_token(token: str, db: Session) -> Optional[User]:
+        api_key: API key string
+
+    Returns:
+        User object if API key is valid, None otherwise
+    """
+    if not api_key:
+        return None
+
+    key_hash = hashlib.sha256(api_key.encode()).hexdigest()
+    api_key_record = db.query(APIKey).filter(
+        APIKey.key_hash == key_hash,
+        APIKey.is_active == True,
```

Ruff (0.14.8)

Avoid equality comparisons to `True`; use `APIKey.is_active:` for truth checks

Replace with `APIKey.is_active`

(E712)